### PR TITLE
feat: kratos hydra integration

### PR DESCRIPTION
**Pull request by grantzvolsky**
This branch is based on the `fix-new-sdk` branch which was ahead of master.

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got green light (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation within the code base (if appropriate).

### CODE REVIEW

File: views/login.hbs [hunk 1]
```diff
@@ -4,6 +4,9 @@
       <h2 class="typography-h2 card-title">Confirm Action</h2>
     {{else if (eq requested_aal "aal2")}}
       <h2 class="typography-h2 card-title">Two-Factor Authentication</h2>
+    {{else if hydra_login_request}}
+      <h2 class="typography-h2 card-title">Sign in</h2>
+      <p>To authenticate {{#if hydra_login_request.client.client_name}}{{hydra_login_request.client.client_name}}{{else}}{{hydra_login_request.client.client_id}}{{/if}}</p>
```
Comment (grantzvolsky) [inline views/login.hbs:9]: Consider linking to client_uri if set.

File: views/registration.hbs [hunk 2]
```diff
@@ -1,6 +1,9 @@
 <div class="auth app-container" id="signup">
   <div class="card">
     <h2 class="typography-h2 card-title">Create an account</h2>
+    {{#if hydra_login_request}}
+      <p>To authenticate {{#if hydra_login_request.client.client_name}}{{hydra_login_request.client.client_name}}{{else}}{{hydra_login_request.client.client_id}}{{/if}}</p>
```
Comment (grantzvolsky) [inline views/registration.hbs:5]: Consider linking to client_uri if set.

File: src/pkg/ui.ts [hunk 3]
```diff
@@ -14,7 +14,7 @@ export const toUiNodePartial = (node: UiNode) => {
   } else if (isUiNodeImageAttributes(node.attributes)) {
     return "ui_node_image"
   } else if (isUiNodeInputAttributes(node.attributes)) {
-    switch (node.attributes.type) {
+    switch (node.attributes && node.attributes['type']) {
```
Comment (aeneasr) [inline src/pkg/ui.ts:17]:
```suggestion
    switch (node.attributes && node.attributes.type) {
```

**Review by aeneasr — CHANGES_REQUESTED** [review]
Just a nit-pick, otherwise LGTM :)

**Comment by jpollard-cs** [comment]
🥳

**Pull request merged**